gqcp/include/Operator/SecondQuantized/USQOneElectronOperator.hpp

```diff
@@ -0,0 +1,445 @@
+        }
+
+        return USQOneElectronOperator<Scalar, 1> {this->fs_alpha[i], this->fs_beta[i]};
+    }
+
+
+    /*
+     *  PUBLIC METHODS
+     */
+
+    /**
+     *  @return read-only matrix representations of all the alpha parameters (integrals) of the different components of this second-quantized operator
+     */
+    const std::array<QCMatrix<Scalar>, Components>& allAlphaParameters() const {
+        return this->fs_alpha;
+    }
+
+
+    /**
+     *  @return writable matrix representations of all the alpha parameters (integrals) of the different components of this second-quantized operator
+     */
+    std::array<QCMatrix<Scalar>, Components>& allAlphaParameters() {
+        return this->fs_alpha;
+    }
+
+    /**
+     *  @return the dimension of the alpha matrices
+     */
+    size_t alphaDimension() const { return this->fs_alpha[0].dimension(); }
+
+
+     /**
+     *  @return read-only matrix representations of all the beta parameters (integrals) of the different components of this second-quantized operator
+     */
+    const std::array<QCMatrix<Scalar>, Components>& allBetaParameters() const {
+        return this->fs_beta;
+    }
+
+
+    /**
+     *  @return writable matrix representations of all the beta parameters (integrals) of the different components of this second-quantized operator
+     */
+    std::array<QCMatrix<Scalar>, Components>& allBetaParameters() {
+        return this->fs_beta;
+    }
+    /**
+     *  @return the dimension of the beta matrices
+     */
+    size_t betaDimension() const { return this->fs_beta[0].dimension(); }
+
+    /**
+     *  @param D                the 1-RDM that represents the wave function
+     *
+     *  @return the expectation values of all components of the one-electron operator
+     */
+    Vector<Scalar, Components> calculateExpectationValue(const OneRDM<Scalar>& D_alpha, const OneRDM<Scalar>& D_beta) const {
+
+        if (this->fs_alpha[0].dimension() != D_alpha.dimension() || this->fs_beta[0].dimension() != D_beta.dimension()) {
+            throw std::invalid_argument("USQOneElectronOperator::calculateExpectationValue(const OneRDM<Scalar>, OneRDM<Scalar>): The given 1-RDM is not compatible with the one-electron operator.");
+        }
+
+        std::array<Scalar, Components> expectation_values_alpha {};  // zero initialization
+        std::array<Scalar, Components> expectation_values_beta {};  // zero initialization
+        for (size_t i = 0; i < Components; i++) {
+            expectation_values_alpha[i] = (this->alphaParameters(i) * D_alpha).trace();
+            expectation_values_beta[i] = (this->betaParameters(i) * D_beta).trace();
+        }
+
+        return Eigen::Map<Eigen::Matrix<Scalar, Components, 1>>(expectation_values_alpha.data()), Eigen::Map<Eigen::Matrix<Scalar, Components, 1>>(expectation_values_beta.data());  // convert std::array to Vector
+    }
+
+
+    /**
+     *  @return the dimension of the total generalized coefficient matrix
```

Comment:
```suggestion
     *  @return the sum of the alpha and beta dimensions
```

Since we've gone through great lengths decoupling coefficients and matrix elements, I feel it's better to avoid this relation in the documentation.